src/AcceptanceTests/Routing/When_scaling_out_senders_that_uses_callbacks.cs

```diff
@@ -0,0 +1,139 @@
+﻿namespace NServiceBus.Azure.Transports.WindowsAzureServiceBus.AcceptanceTests.Routing
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.AzureServiceBus;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NServiceBus.Settings;
+    using NUnit.Framework;
+
+    public class When_scaling_out_senders_that_uses_callbacks : NServiceBusAcceptanceTest
+    {
+        const int numberOfMessagesToSend = 5;
+
+        [Test]
+        public async Task Should_only_deliver_response_to_one_of_the_instances()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<ServerThatRespondsToCallbacks>()
+                .WithEndpoint<ScaledOutClient>(b =>
+                {
+                    b.CustomConfig(c =>
+                    {
+                        c.ScaleOut().InstanceDiscriminator("A");
+                        c.GetSettings().Set("Client", "A");
+                    });
+                    b.When(async (bus, c) =>
+                    {
+                        for (var i = 0; i < numberOfMessagesToSend; i++)
+                        {
+                            var sendOptions = new SendOptions();
+                            sendOptions.RouteReplyToThisInstance();
+                            await bus.Send(new MyRequest()
+                            {
+                                Client = "A"
+                            }, sendOptions);
+                        }
+                    });
+                })
+                .WithEndpoint<ScaledOutClient>(b =>
+                {
+                    b.CustomConfig(c =>
+                    {
+                        c.ScaleOut().InstanceDiscriminator("B");
+                        c.GetSettings().Set("Client", "B");
+                    });
+                    b.When(async (bus, c) =>
+                    {
+                        for (var i = 0; i < numberOfMessagesToSend; i++)
+                        {
+                            var sendOptions = new SendOptions();
+                            sendOptions.RouteReplyToThisInstance();
+                            await bus.Send(new MyRequest()
+                            {
+                                Client = "B"
+                            }, sendOptions);
+                        }
+                    });
+                })
+                .Done(c => c.RepliesReceived >= numberOfMessagesToSend * 2)
```

Comment:
Wait. If message could fail and test counts also retries (and it does) it could fail because `RepliesReceived` is greater than `numberOfMessagesToSend * 2`